src/brogue/Items.c

```diff
@@ -3969,6 +3969,8 @@ boolean imbueInvisibility(creature *monst, short duration) {
     if (monst && !(monst->info.flags & (MONST_INANIMATE | MONST_INVISIBLE | MONST_INVULNERABLE))) {
         if (monst == &player || monst->creatureState == MONSTER_ALLY) {
             autoID = true;
+        } else if (monsterRevealed(monst)) {
```

Comment:
This is no longer clearly sufficient to autoID since if we hit a revealed but unseen creature (a psychic emanation), the bolt could also be domination, empowerment, slowing, plenty.. I get this is gated by MONST_INVISIBLE which accounts for the most common cause of revealed-unseen (maybe the only case where it can be hit by a bolt?), but I think we should be careful. What are your thoughts?

```suggestion
        } else if (canSeeMonster(monst) && monsterRevealed(monst)) {
```

Reply:
might this need to be `canSeeMonsterDirectly`? if the player is telepathic and zaps a hidden monster (underwater or under a bog) the monster will continue to be visible as a psychic emanation

Reply:
i actually need to confirm you can zap submerged monsters, i can't remember

Reply:
Even if it were possible, I don't think so. Hidden monsters are not seen, by definition of canSeeMonster.